src/library/scala/collection/immutable/VectorMap.scala

```diff
@@ -26,140 +25,194 @@ import scala.annotation.unchecked.{uncheckedVariance => uV}
   *  @tparam V      the type of the values associated with the keys in this vector map.
   *
   * @author Matthew de Detrich
+  * @author Odd Möller
   * @version 2.13
   * @since 2.13
   * @define coll immutable vector map
   * @define Coll `immutable.VectorMap`
   */
-final class VectorMap[K, +V] private[immutable] (
-    private[immutable] val fields: Vector[K],
-    private[immutable] val underlying: Map[K, (Int, V)])
-    extends AbstractMap[K, V]
+final class VectorMap[K, +V] private (
+    private[immutable] val fields: Vector[Any],
+    private[immutable] val underlying: Map[K, (Int, V)], dummy: Boolean)
```

Comment:
What's the purpose of the `dummy` parameter?

Reply:
I imagine to disambiguate with the secondary constructor

```scala
  private[immutable] def this(fields: Vector[K], underlying: Map[K, (Int, V)]) = {
    this(fields, underlying, false)
  }
```

Reply:
@szeiger, @hrhino Yes, thats right, the dummy parameter is there to make the erased signatures of the primary and secondary constructors different from each other. Only the secondary constructor having the ostensible correct type of the fields vector should be available to other members of the `immutable` package, while the primary constructor using the relaxed type of the fields vector should only be available to the `VectorMap` class itself.